### PR TITLE
Use get search filter based on identity

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
Using the method `getSearchFilterBasedOnIdentity` from `Auth` package to add pre SearchFilters to these 4 operations: `search-type`, `history-type`, `search-system`, `history-system`.

This allows us to only search for resources that a requester has access to.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/router/handlers/resourceHandler.ts [hunk 1]
```diff
@@ -71,11 +86,17 @@ export default class ResourceHandler implements CrudHandlerInterface {
         );
     }
 
-    async typeHistory(resourceType: string, queryParams: any) {
+    async typeHistory(resourceType: string, queryParams: any, userIdentity: KeyValueMap) {
```
Comment (rsmayda) [inline src/router/handlers/resourceHandler.ts:89]: Don't forget instanceHistory

Comment (nguyen102) [reply src/router/handlers/resourceHandler.ts:89]: For `getSearchFilterBasedOnIdentity` it only includes `operation: 'search-type' | 'search-system' | 'history-type' | 'history-system';`

Should we add `history-instance` to that list? That was the original reason why I didn't include `history-instance` in this PR.

Comment (nguyen102) [reply src/router/handlers/resourceHandler.ts:89]: Added this PR to include `history-instance`
https://github.com/awslabs/fhir-works-on-aws-interface/pull/41

Comment (nguyen102) [reply src/router/handlers/resourceHandler.ts:89]: Updated this PR to include filtering `history-instance`

**Comment by nguyen102** [comment]
@carvantes for some reason GH won't let me add you to the reviewer list for this PR. So I'll just tag you here as a reviewer also.

**Review by rsmayda — DISMISSED** [review]

File: src/router/metadata/metadataHandler.test.ts [hunk 2]
```diff
@@ -650,5 +650,9 @@ test('R4: FHIR Config V4 with all productInfo params', async () => {
         fhirVersion: '4.0.1',
         format: ['json'],
         rest: response.resource.rest,
-    });
+    };
+
+    expectedResponse.date = expect.stringMatching(utcTimeRegExp);
+
+    expect(response.resource).toEqual(expectedResponse);
```
Comment (carvantes) [inline src/router/metadata/metadataHandler.test.ts]: You can use `toMatchObject` instead:
```js
expect(response.resource).toMatchObject({
        resourceType: 'CapabilityStatement',
        name: 'product.machine.name',
        title: 'Product Title Capability Statement',
        description: 'Product Description',
        purpose: 'Product Purpose',
        copyright: 'Copyright',
        status: 'active',
        date: expect.stringMatching(utcTimeRegExp),
        publisher: 'Organization Name',
        kind: 'instance',
        software: {
            name: 'Product Title',
            version: '1.0.0',
        },
        implementation: {
            description: 'Product Description',
            url: 'http://example.com',
        },
        fhirVersion: '4.0.1',
        format: ['json'],
        rest: response.resource.rest,
    });
```

**Review by carvantes — DISMISSED** [review]
looks good, just a minor comment on a test

File: src/router/metadata/metadataHandler.test.ts (hunk 2 above)
Comment (nguyen102) [reply src/router/metadata/metadataHandler.test.ts]: updated

**Review by carvantes — APPROVED** [review]

**Pull request merged**